File: 4.8-bc/ts/stage-3-hexagonal/src/catalog/module.ts
// Self-wiring entry point для Catalog BC.
import type { DbPool } from "../shared/db.js";

import { Service } from "./app/service.js";
import { Handler } from "./infra/http/handler.js";
import { PgProductRepo } from "./infra/postgres/productRepo.js";

export interface Module {
  service: Service;
  handler: Handler;
}

export function newModule(db: DbPool): Module {
  const repo = new PgProductRepo(db);
  const service = new Service(repo);
  const handler = new Handler(service);
  return { service, handler };
}

File: 4.8-bc/py/stage-3-hexagonal/app/catalog/module.py
"""Catalog BC self-wiring."""
from __future__ import annotations

from dataclasses import dataclass

from fastapi import APIRouter
from sqlalchemy.ext.asyncio import async_sessionmaker

from app.catalog.app.service import Service
from app.catalog.infra.http.handler import build_router
from app.catalog.infra.postgres.product_repo import ProductRepo


@dataclass
class Module:
    service: Service
    _router: APIRouter

    def routes(self) -> APIRouter:
        return self._router


def New(session_maker: async_sessionmaker) -> Module:
    repo = ProductRepo(session_maker)
    svc = Service(repo)
    router = build_router(svc)
    return Module(service=svc, _router=router)
